docusaurus/docs/dev-docs/backend-customization/controllers.md

```diff
@@ -30,99 +30,159 @@ A new controller can be implemented:
-    // validateQuery (optional)
-    // to throw an error on query params that are invalid or the user does not have access to
-    await this.validateQuery(ctx);
-
-    // sanitizeQuery to remove any query params that are invalid or the user does not have access to
-    // It is strongly recommended to use sanitizeQuery even if validateQuery is used
-    const sanitizedQueryParams = await this.sanitizeQuery(ctx);
-    const { results, pagination } = await strapi.service('api::restaurant.restaurant').find(sanitizedQueryParams);
-    const sanitizedResults = await this.sanitizeOutput(results, ctx);
-
-    return this.transformResponse(sanitizedResults, { pagination });
-  }
-}));
+const { createCoreController } = require("@strapi/strapi").factories;
+
+module.exports = createCoreController(
+  "api::restaurant.restaurant",
+  ({ strapi }) => ({
+    /**
+     * Example 1: Modifying a Strapi controller function
+     *
+     * If you need to modify the input or output of a pre-defined Strapi controller method,
+     * write a method of the same name, and use `super` to call the parent method.
+     * */
+    async find(ctx) {
+      // your custom logic for modifying the input
+      ctx.query = { ...ctx.query, locale: "en" }; // force ctx.query.locale to 'en' regardless of what was requested
+
+      // Call the default parent controller action
+      const result = await super.find(ctx);
+
+      // your custom logic for modifying the output
+      result.meta.date = Date.now(); // change the date that is returned
+
+      return result;
+    },
+
+    /**
+     * Example 2: Replacing a Strapi controller function
+     *
+     * If you need to completely replace the behavior of a pre-defined Strapi controller method,
+     * you can do so by simply implementing a method of the same name.
+     *
+     * Caution: You will need to manage the security of the request and results on your own,
+     * as demonstrated in this example.
+     * */
+    async find(ctx) {
+      // validateQuery throws an error if any of the query params used are inaccessible to ctx.user
+      // That is, trying to access private fields, fields they don't have permission for, wrong data type, etc
+      await this.validateQuery(ctx);
+
+      // sanitizeQuery silently removes any query params that are invalid or the user does not have access to
+      // It is recommended to use sanitizeQuery even if validateQuery is used, as validateQuery allows
+      // a number of non-security-related cases such as empty objects in string fields to pass, while sanitizeQuery
+      // will remove them completely
+      const sanitizedQueryParams = await this.sanitizeQuery(ctx);
+
+      // Perform whatever custom actions are needed
+      const { results, pagination } = await strapi
+        .service("api::restaurant.restaurant")
+        .find(sanitizedQueryParams);
+
+      // sanitizeOutput removes any data that was returned by our query that the ctx.user should not have access to
```

Comment:
Should we also mention removing private fields and passwords? Or was it inferred by `ctx.user should not have access t`?

Reply:
@innerdvations If you'd like to update this, feel free to do so. 
But we'll have improved sanitization docs for v5 anyway (see yesterday's discussion on Slack).

Reply:
I don't think it's necessary, this is the "custom controller" section we just need to point out why it's important, and if they want a complete explanation they can look in the sanitization section.